extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -32,17 +33,34 @@ export class AnalysesResultsManager {
     void this.logger.log(`Downloading and processing results for ${analysisSummary.nwo}`);
 
     await this.downloadSingleAnalysisResults(analysisSummary, credentials);
+    await publishResults(this.analysesResults);
   }
 
-  public async downloadAllResults(
-    analysisSummaries: AnalysisSummary[],
+  public async downloadAnalysesResults(
+    analysesToDownload: AnalysisSummary[],
+    token: CancellationToken | undefined,
+    publishResults: (analysesResults: AnalysisResults[]) => Promise<void>
   ): Promise<void> {
     const credentials = await Credentials.initialize(this.ctx);
 
-    void this.logger.log('Downloading and processing all results');
+    void this.logger.log('Downloading and processing analyses results');
 
-    for (const analysis of analysisSummaries) {
-      await this.downloadSingleAnalysisResults(analysis, credentials);
+    const batchSize = 3;
+
+    for (let i = 0; i < analysesToDownload.length; i += batchSize) {
+      if (token && token.isCancellationRequested) {
+        throw new Error('Downloading of analyses results has been cancelled');
```

Comment:
There is a `UserCancellationException` for this purpose. Set the `silent` flag to true if you don't want a pop-up.

Reply:
Thanks I wasn't aware of that!